<!DOCTYPE html>
<html lang="en">

    <head>
        <meta charset="UTF-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>Document</title>
    </head>

    <body>

    </body>
    <script>
        async function fn(){
            try {
                console.log(111);
                let data=await Promise.reject("cuowu")
                // await后面一旦是一个reject:Promise.reject("cuowu"),
                // 就会抛错,会卡住try里的代码
                // let data =await new Promise((resolve,reject)=>{
                //     setTimeout(() => {
                //         reject("错误")
                //     }, 1000);
                // })
                console.log(222);
            } catch (error) {
                console.log(error);
            }
            console.log(333);
        }
        fn()
    </script>

</html>